[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26390 [quant][graph] PyTorch Graph Mode Quantization API
* **#26389 [quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant**
* #26425 [quant][graphmode] Get scalar type from observer module

Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17504458](https://our.internmc.facebook.com/intern/diff/D17504458)

jerryzh168 committed:
[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant

Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -586,9 +586,16 @@ void InsertQuantDeQuantImpl(
-    const QConfigDict& qconfig_dict) {
+    const QConfigDict& qconfig_dict,
+    bool inplace) {
+  script::Module module;
```
Nit: `script::Module module = inplace ? input_module : input_module.clone();`

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -597,8 +604,14 @@ TORCH_API void InsertObservers(
-  script::Module module = input_module.clone();
+    const std::string& method_name,
+    bool inplace) {
+  script::Module module;
```
Same.

ZolotukhinM commented on torch/csrc/jit/passes/quantization.h:
```diff
@@ -61,7 +61,8 @@ TORCH_API void InsertObservers(
         std::string,
-        std::tuple<script::Module, script::Module>>& qconfig_dict);
+        std::tuple<script::Module, script::Module>>& qconfig_dict,
+    bool inplace = false);
```
The function needs to return the cloned module. Also, please update the comment correspondingly.

ZolotukhinM reviewed (approved):
Overall looks good modulo couple of remarks (see inline)!

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17504458](https://our.internmc.facebook.com/intern/diff/D17504458)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17504458](https://our.internmc.facebook.com/intern/diff/D17504458)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graph] Add inplace argument to InsertObservers and InsertQuantDeQuant"


Summary:
att

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17504458](https://our.internmc.facebook.com/intern/diff/D17504458)

[ghstack-poisoned]
